packages/react/src/components/List/ListItem/ListItem.jsx

```diff
@@ -328,7 +331,7 @@ const ListItem = ({
                 [`${iotPrefix}--list-item--category`]: isCategory,
                 [`${iotPrefix}--list-item--content--values__disabled`]: disabled,
               })}
-              title={value}
+              title={title ?? value}
```

Comment:
I would replace `??` with `||` since `title` may come from prop as an empty string and value will be ignored.